doc/adr/0004-git-branching-strategy.md

```diff
@@ -0,0 +1,86 @@
+# ADR-0004. Git Branching Strategy
+
+Date: 12/11/2019
+
+## Status
+
+Accepted
+
+## Context
+
+A team utilising git repositories should have a consistent set of branching
+strategies. This will enable encoding relevant information into a branch name
+for improved efficiency. Each component of the branch name will then be able to
+be utilised in other areas of automation.
+
+## Decision
+
+Each branch _**will**_ have it's own deployed environment to use for testing the
+developed code. There will be a 1 to 1 relationship between deployed
+environments and repository brnaches. The only exception to this 1 to 1
+relationship will be the _**master**_ branch. The _**master**_ branch may be used to
+back any environment that needs to be live like.
```

Comment:
Could we move this block below the branching strategy. It feels like the deployment of environments is secondary to the approach, and is more a consequence of having a consistent approach (i.e. we could adopt this strategy and take a completely different approach to managing the dev subscription, though I'm not suggesting we do)